### ISSUE TITLE
Python tags error message

### ISSUE THREAD

**Issue by elextr** [geany/geany#357]
In 1.24.1 I intermittently get errors when editing Python "geany: Warning: ignoring null tag in <the filename>".

Of course being intermittent they never show when I am watching the terminal, only when it is covered by the Geany window. :(

So I can't add any information on what the buffer needs to contain to trigger it.

This issue is just a placeholder so hopefully anyone else who notices will add more info.

**Comment by erm3nda** [comment]
I have same thing on the 1.25. I see it because Geany took a lot of time to show up the window, so I tried it from terminal and saw that.

**Comment by elextr** [comment]
@erm3nda so can you provide more info about what triggers it?

**Comment by erm3nda** [comment]
The only think i can remember, is that I compiled it from source.
This only happens when I open a file with Geany, so it's something at program startup.

I found it just now. Deactivating **Geanyprj** addon.
This addon does scan every /home folder to search for tags, so bad.

**Comment by elextr** [comment]
@erm3nda well, that says some file that it is scanning is the cause of the problem, can you see if you can find which one and tell us what causes it, so we can fix the parser.  Its probably some illegal syntax in the file or something new the parser doesn't understand.  Since there are not queues of people complaining its obviously something rare, so it would be good if you can help to identify it since you seem to have a file with the problem.

**Comment by erm3nda** [comment]
I only have error on .js files, files from example Wordpress install.
Example: geany: Warning: ignoring null tag in /home/m3nda/test/wp-bash/wordpress/wp-includes/js/zxcvbn.min.js

I have dozens of them, so i don't think it's a particular wrong syntax... more like a bug on the lexer.

Quote: Since there are not queues of people complaining its obviously something rare
I think most of them just uninstalled it while being SLOOOOOOOW :)

**Comment by elextr** [comment]
ok, I got it on a Python file (according to the OP, I can't remember now, its over a year ago :)

Is there one of the js files that is small, exhibits the problem and that you don't mind sharing that you could put in a gist?

**Comment by erm3nda** [comment]
I've pasted one of the list at http://pastebin.com/Cbc73X0V, but, you can get the same file donwloading a up-to-date wordpress.

**Comment by codebrainz** [comment]
> I only have error on .js files, wordpress files as example:
>> geany: Warning: ignoring null tag in /home/m3nda/test/wp-bash/wordpress/wp-includes/js/zxcvbn.min.js

Does it cause same issue on a non-minified version of the same file?

**Comment by erm3nda** [comment]
@codebrainz So smart question. I've noticed that EVERY *.js file where a *-min.js version so the problem is only about minified *.js files in my case.

**Comment by codebrainz** [comment]
> I think most of them just uninstalled it while being SLOOOOOOOW :)

If it's always minified, this makes sense. Sinctilla/Geany is notorious for not handling single-long-line files, especially with line wrapping enabled.

**Comment by elextr** [comment]
Since the original post is about a Python problem, it is unlikely that the minified JS issue is the same problem, please move it to another issue if its considered still a problem.

**Comment by codebrainz** [comment]
Yeah, likely two different issues manifesting with the same debug output.

**Comment by kjwein** [comment]
I am using Geany-1.27 and I started noticing the same issue when editing a python file. I could replicate the issue when editing a python class with a function definition inside. If you define a function but do not name it like this
```python
def (self, args)
```
anytime you type anything else in the file, I get the warning in the terminal window. As soon as you complete the function name,

```python
def function(self, args):
```
 the warning stops printing to screen. The function does not have to be in a class. Hope this helps!

**Comment by elextr** [comment]
Confirmed with 1.29 Git.  

@techee the fix is probably in the Python parser not to generate null symbols, tried Pascal and Rust with null function names (both languages that have a syntactic indication of function declaration) and no error.  I guess fixing the parser means upstream.  But possibly tagmanager should ignore not complain either.

**Comment by techee** [comment]
The warning comes from ctags:

https://github.com/geany/geany/blob/master/ctags/main/entry.c#L390

But it really is just a warning and I'd say it's useful to know if something is wrong. And @b4n wrote a new token-based Python parser which is already merged in uctags and and which might have fixed this already. We should backport it to Geany after #1160.

**Comment by elextr** [comment]
@b4n can consider himself pre-pinged if the new parser still has the problem :)

**Comment by codebrainz** [comment]
IMO if the new Python lexer is far off at all, it might be good to squelch that warning as it is just noise in the context of someone typing a necessarily - at times - broken source file.

**Comment by techee** [comment]
> IMO if the new Python lexer is far off at all, it might be good to squelch that warning as it is just noise in the context of someone typing a necessarily - at times - broken source file.

Hopefully not far - I believe #1160 is ready to be merged and then the new Python parser can be just copied with minimal changes.

Upstream ctags also reports line number in the warning so it's more descriptive (should get backported to Geany). Null tag names shouldn't be generated no matter what the user has typed and this message helps to detect an error in the parser which would otherwise be undetected.

**Comment by techee** [comment]
By the way the NULL tag warning for javascript has been fixed upstream so it's a matter of getting the new version from there.
